File: Lista 02/elevador.cpp
#include <iostream>
using namespace std;

class elevador{
    private:
        int capacidade, totalAndares, atual, pessoas;
    public:
        elevador(int totalAndares = 14, int capacidade = 6){
            pessoas = 0;
            atual = 0;
            this->totalAndares = totalAndares;
            this->capacidade = capacidade;
        }

        int mudarAndar(){
            int maisGente, mudar, destino;
            std::cout<<"Você deseja mudar de andar?\n"
            <<"1- Sim\n"
            <<"2- Não"<<endl;
            cin>>mudar;
            if (mudar < 0 || mudar > 14){
                std:cout<<"Por favor, selecione um andar válido"<<endl;
                return mudarAndar();
            };
            std::cout<<"Quantas pessoas vão entrar no elevador? "<<endl;
            cin>>maisGente;
            if (pessoas + maisGente > capacidade){
                std::cout<<"O elevador não consegue comportar esse número de pessoas a mais"<<endl;
                return 0;
            }
            pessoas += maisGente;
            if (mudar == 1){
                std::cout<<"Para qual andar você vai? "<<endl;
                cin>>destino;
                if (destino > 14){
                    std::cout<<"Por favor, escolha um andar menor que 14"<<endl;
                    return 0;
                };
                atual = destino;
                std::cout<<"Você chegou ao "<<destino<<" andar"<<endl;
                return 1;
            }
            cout<<"Por favor, caso não deseje mudar de andar deixe o elevador"<<endl;
            return 0;
        };

};

File: Lista 02/Elevador.java
public class Elevador {
    private int capacidade;
    private int totalAndares;
    private int atual;
    private int pessoas;

    // Construtor
    public Elevador(int totalAndares, int capacidade) {
        this.pessoas = 0;
        this.atual = 0;
        this.totalAndares = totalAndares;
        this.capacidade = capacidade;
    }

    // Métodos para mudar andar
    public boolean mudarAndar(int destino, int maisGente) {
        if (destino < 0 || destino > totalAndares) {
            System.out.println("Por favor, escolha um andar válido.");
            return false;
        }
        if (pessoas + maisGente > capacidade) {
            System.out.println("O elevador não consegue comportar esse número de pessoas a mais.");
            return false;
        }
        pessoas += maisGente;
        atual = destino;
        System.out.println("Você chegou ao andar " + destino);
        return true;
    }

    // Getters
    public int getCapacidade() {
        return capacidade;
    }

    public int getTotalAndares() {
        return totalAndares;
    }

    public int getAtual() {
        return atual;
    }

    public int getPessoas() {
        return pessoas;
    }

    // Setters
    public void setCapacidade(int capacidade) {
        if (capacidade > 0) {
            this.capacidade = capacidade;
        } else {
            System.out.println("Capacidade inválida!");
        }
    }

    public void setTotalAndares(int totalAndares) {
        if (totalAndares > 0) {
            this.totalAndares = totalAndares;
        } else {
            System.out.println("Total de andares inválido!");
        }
    }

    public void setAtual(int atual) {
        if (atual >= 0 && atual <= totalAndares) {
            this.atual = atual;
        } else {
            System.out.println("Andar inválido!");
        }
    }

    public void setPessoas(int pessoas) {
        if (pessoas >= 0 && pessoas <= capacidade) {
            this.pessoas = pessoas;
        } else {
            System.out.println("Número de pessoas inválido!");
        }
    }
}
